pkg/executor/join/outer_join_probe.go

```diff
@@ -110,37 +116,41 @@ func (j *outerJoinProbe) ScanRowTable(joinResult *hashjoinWorkerResult, sqlKille
 	if joinResult.chk.IsFull() {
 		return joinResult
 	}
-	if j.rowIter == nil {
-		panic("scanRowTable before init")
-	}
+
 	j.cachedBuildRows = j.cachedBuildRows[:0]
 	meta := j.ctx.hashTableMeta
 	insertedRows := 0
 	remainCap := joinResult.chk.RequiredRows() - joinResult.chk.NumRows()
+
 	for insertedRows < remainCap && !j.rowIter.isEnd() {
-		currentRow := j.rowIter.getValue()
-		if !meta.isCurrentRowUsed(currentRow) {
-			// append build side of this row
-			j.appendBuildRowToCachedBuildRowsAndConstructBuildRowsIfNeeded(createMatchRowInfo(0, currentRow), joinResult.chk, 0, false)
-			insertedRows++
-		}
+		j.appendBuildRow(meta, joinResult.chk, j.rowIter.getValue(), &insertedRows)
```

Comment:
why change this

Reply:
> why change this

I think it's more readable to put these codes into a function and reader could know the purpose of these codes from function name.